msal/src/main/java/com/microsoft/identity/client/internal/controllers/LocalMSALController.java

```diff
@@ -116,6 +125,17 @@ private AuthorizationResult performAuthorizationRequest(final OAuth2Strategy str
         return result;
     }
 
+    private AuthorizationAgent validAuthorizationAgent(final AuthorizationAgent agent, final Context context) {
```

Comment:
Looks like it's possible to move this logic inside the factory.  Feels like it's a responsibility of the factory to return an authorization strategy that will work on the current device.